### PR TITLE
wasm-objdump: Report data segment names

### PR DESCRIPTION
**Pull request by sbc100**
Also fix ordering of data names and table segment names.

### CODE REVIEW

File: src/binary-reader-objdump.cc [hunk 1]
```diff
@@ -650,14 +653,28 @@ Result BinaryReaderObjdumpDisassemble::OnOpcodeIndexIndex(Index value,
...
 
 Result BinaryReaderObjdumpDisassemble::OnOpcodeUint32Uint32(uint32_t value,
                                                             uint32_t value2) {
   Offset immediate_len = state->offset - current_opcode_offset;
-  LogOpcode(immediate_len, "%u %u", value, value2);
+  string_view name;
+  if (current_opcode == Opcode::MemoryInit &&
+      !(name = GetSegmentName(value)).empty()) {
+    LogOpcode(immediate_len, "%d <" PRIstringview ">", value,
+              WABT_PRINTF_STRING_VIEW_ARG(name));
```
Comment (tlively) [inline src/binary-reader-objdump.cc:674]: Looks like `value2` is missing here.

File: test/dump/extended-names.txt [hunk 2]
```diff
@@ -92,82 +94,82 @@
...
-000007f: 08                                        ; FIXUP subsection size
-0000088: 08                                        ; name subsection type
-0000089: 00                                        ; subsection size (guess)
-000008a: 01                                        ; num names
-000008b: 00                                        ; elem index
-000008c: 05                                        ; string length
-000008d: 656c 656d 31                             elem1  ; elem name 0
-0000089: 08                                        ; FIXUP subsection size
-000004a: 47                                        ; FIXUP section size
-0000089: warning: out-of-order sub-section
```
Comment (tlively) [inline test/dump/extended-names.txt:170]: Where did this warning go?

File: test/dump/extended-names.txt [hunk 3]
```diff
@@ -204,10 +208,12 @@ Custom:
  - memory[0] <mem2>
  - global[0] <g1>
  - global[1] <g2>
+ - elemseg[0] <elem1>
```
Comment (tlively) [inline test/dump/extended-names.txt:211]: Where does this "elem1" come from?

File: test/dump/extended-names.txt (hunk 2 above)
Comment (sbc100) [reply test/dump/extended-names.txt:170]: The order was wrong in `src/binary-writer.cc` .. this change has a fix that that.  Want me to split it out?

File: test/dump/extended-names.txt (hunk 3 above)
Comment (sbc100) [reply test/dump/extended-names.txt:211]: The out of order name section meant is wasn't being displayed here.

File: src/binary-reader-objdump.cc (hunk 1 above)
Comment (sbc100) [reply src/binary-reader-objdump.cc]: Good point.
Fixed.

**Review by tlively — APPROVED** [review]

File: test/dump/extended-names.txt (hunk 2 above)
Comment (tlively) [reply test/dump/extended-names.txt:170]: Ah, got it, thanks! Keeping it in seems fine.

**Pull request merged**